src/sql/src/plan/query.rs

```diff
@@ -1632,49 +1632,60 @@ fn plan_view_select(
             allow_windows: false,
         };
         let mut group_key = vec![];
-        let mut group_exprs = vec![];
+        let mut group_exprs: HashMap<HirScalarExpr, ScopeItem> = HashMap::new();
```

Comment:
mm the EXPLAIN output is a good consideration. To maintain the existing order, and since BTreeMap would require HirScalarExpr to be Ord, I opted to build up a vec of the HIR exprs similar to before, and then iterate through the HashMap in that order.

and as a bonus, restoring the order fixed all the test failures 😄

Reply:
Makes sense and sounds good to me!